### ISSUE TITLE
`PLC0414` false positive for stubs

### ISSUE THREAD

**Issue by Avasam** [charliermarsh/ruff#3734]
`PLC0414` should not be applied to `.pyi`. Explicit re-exports are a valid convention to indicate that a type is meant to be exposed as re-exported, and isn't just used in the stub for typing reference.

https://github.com/python/typeshed/blob/main/CONTRIBUTING.md#conventions
> Imports in stubs are considered private (not part of the exported API) unless:
> - they use the form from library import name as name (sic, using explicit as even if the name stays the same); or
> - they use the form from library import * which means all names from that library are exported.

For example:
```py
from ._modules import controller as controller, event as event, gui as gui, memory as memory, savestate as savestate
```

**Comment by charliermarsh** [comment]
It looks like Pylint _does_ flag these in `.pyi` files so this would be a deviation.

I'm actually kind of surprised this rule exists at all any more. Isn't explicit re-export part of the de facto public / private standard used by type checkers? E.g., see [Pyright](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface).

**Comment by Avasam** [comment]
Pylint does not support type stubs: https://github.com/PyCQA/pylint/issues/4987

**Comment by charliermarsh** [comment]
Ah yeah. I mean, we definitely could disable this rule in `.pyi` files. It's just that the logic could also apply to `.py` files. But if no one would ever want this enforced in a `.pyi` file (but some _would_ want it enforced in `.pyi` files), then I'd err on the side of disabling it in `.pyi` files.

**Comment by charliermarsh** [comment]
Figured why not fix, better than flagging the `.pyi` files.
